### DISCUSSION TITLE
Tesseract OCR

### DISCUSSION THREAD

**Discussion by smokris** [phiresky/ripgrep-all#203] [Show your adapter]
> _Tesseract is an open source [text recognition (OCR)](https://en.wikipedia.org/wiki/Optical_character_recognition) Engine, available under the [Apache 2.0 license](http://www.apache.org/licenses/LICENSE-2.0). … Tesseract can be used directly via [command line](https://tesseract-ocr.github.io/tessdoc/Command-Line-Usage.html), or (for programmers) by using an [API](https://github.com/tesseract-ocr/tesseract/blob/main/include/tesseract/baseapi.h) to extract printed text from images. It supports a [wide variety of languages](https://tesseract-ocr.github.io/tessdoc/Data-Files-in-different-versions.html)._ — https://tesseract-ocr.github.io/tessdoc/

Tesseract installation instructions for various platforms: https://tesseract-ocr.github.io/tessdoc/Installation.html#platforms

ripgrep-all config:
```json
{
  "$schema": "./config.schema.json",
  "custom_adapters": [
    {
      "name": "tesseract",
      "version": 1,
      "description": "Uses Tesseract OCR to recognize text in image files",
      "extensions": ["jpg", "jpeg", "jpe", "png", "webp", "gif", "tif", "tiff", "jp2", "j2k", "jpf", "jpm", "jpg2", "jpg2000", "jpeg2000", "j2c", "jpc", "jpx", "bmp", "pnm"],
      "mimetypes": ["image/jpeg", "image/png", "image/webp", "image/gif", "image/tiff", "image/jp2", "image/jpx", "image/jpm", "image/bmp", "image/x-portable-anymap"],
      "binary": "tesseract",
      "args": ["--psm", "1", "-", "-"],
      "disabled_by_default": false,
      "match_only_by_mime": false
    }
  ]
}
```

**Comment by sigmaSd** [comment]
2026 update:
This needs to be put in ~/.config/ripgrep-all/config.jsonc 
For some reason .json is not picked up
Also the readme is confusing it mentions being able to search jpg, but without this config it wont 
Tip: run with --debug to see the issue
